### PR TITLE
Setup S3 and implement upload functionality

### PR DESCRIPTION
**Pull request by michaeldotdev**
### What this PR does (required):

1. Setup AWS S3 Bucket
2. Upload file to Express server then pushes to S3 Bucket

### Screenshots / Videos (required):
Screen Shot 2021-09-15 at 1 31 02 PM
Screen Shot 2021-09-15 at 2 51 21 PM

### Any information needed to test this feature (required):

1. Change directory into server
2. npm run dev
3. Using Postman to send a POST request to localhost:3001/profilePhotos attaching an image to the body

### Any issues with the current functionality (optional):
N/A

### CODE REVIEW

File: server/S3.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY;
+
+const s3 = new S3({
+  region,
+  accessKeyId,
+  secretAccessKey,
+})
+
+// Function to handle uploading to S3 bucket
+function uploadPhoto(file) {
```
Comment (sundayezeilo) [inline server/S3.js:17]: This should be an `async` `function`, since it returns a `Promise`

File: server/app.js [hunk 2]
```diff
@@ -56,6 +60,13 @@ if (process.env.NODE_ENV === "production") {
   });
 }
 
+// POST request for uploading to Express server then pushing to S3 Bucket
+app.post('/profilePhotos', upload.single('profilePhoto'), async (req, res) => {
+  const file = req.file;
+  const result = await uploadPhoto(file);
+  res.send('upload complete!')
```
Comment (sundayezeilo) [inline server/app.js:67]: Verify that upload was successful before returning a success response, if not, return a failure status.

**Review by sundayezeilo — COMMENTED** [review]
Good work here! Just check a few comments that I left.
Happy coding!

File: server/app.js (hunk 2 above)
Comment (michaeldotdev) [reply server/app.js:67]: Got it! Have a .catch to resolve the errors.

File: server/S3.js (hunk 1 above)
Comment (michaeldotdev) [reply server/S3.js:17]: Where that function is called, its async - on app.js line 66.

Comment (rajivtitus) [reply server/S3.js:17]: That works too. However, having this function as `async` would make things easier if we plan to use it in multiple places :)

File: server/app.js (hunk 2 above)
Comment (rajivtitus) [reply server/app.js:67]: Please handle errors here. Also, wouldn't it be better to send the image URL as a response?

**Comment by rajivtitus** [comment]
Closing these PRs as Michael is no longer a part of CAP :)

**Comment by rajivtitus** [comment]
Closing these PRs as Michael is no longer a part of CAP :)

**Pull request closed without merging**